build/tools/Automation/Python/Python.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Oryx.Automation.Client;
+using Microsoft.Oryx.Automation.Models;
+using Microsoft.Oryx.Automation.Python.Models;
+using Microsoft.Oryx.Automation.Services;
+using Microsoft.Oryx.Automation.Telemetry;
+using Newtonsoft.Json;
+using Oryx.Microsoft.Automation.Python;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.NamingConventions;
+
+namespace Microsoft.Oryx.Automation.Python
+{
+    public class Python
+    {
+        private readonly IHttpClient httpClient;
+        private readonly ILogger logger;
+        private readonly IVersionService versionService;
+        private readonly IYamlFileReaderService yamlFileReaderService;
+
+        public Python(IHttpClient httpClient, ILogger logger, IVersionService versionService, IYamlFileReaderService yamlFileReaderService)
+        {
+            this.httpClient = httpClient;
+            this.logger = logger;
+            this.versionService = versionService;
+            this.yamlFileReaderService = yamlFileReaderService;
+        }
+
+        public async Task RunAsync(string oryxRootPath)
```

Comment:
I made a similar comment in the refactor PR about not passing the `oryxRootPath` variable around, but thinking about it a little more, I think the correct route would be the following:

`IYamlFileService` accepts the `oryxRootPath` as a constructor parameter that we'll provide when setting up DI for that interface

```cs
 .AddSingleton<IYamlFileReaderService>(new YamlFileReaderService(oryxRootPath))
```

Then every call in the `YamlFileReaderService` will be in the scope of that provided root path, so we don't need to worry about providing it again as a part of the method signature for every single call.

Similarly, we should introduce another service interface that's responsible for reading/writing the `versionsToBuild.txt` files so that we can roll-out a similar pattern that avoids having to pass around a path.

Reply:
Updated to `YamlFileSerivce` and have introduced a `FileService`. Now `oryxRootPath` does not need to be passed to `RunAsync`